apps/src/music/blockly/FieldTune.ts

```diff
@@ -0,0 +1,217 @@
+      }
+    }
+  }
+
+  getText() {
+    const {notes, instrument} = this.getValue();
+    if (notes.length === 0) {
+      return 'select notes';
+    }
+
+    return `${instrument} (${this.getTruncatedNotes(notes)})`;
+  }
+
+  protected render_() {
+    if (this.backgroundElement) {
+      this.backgroundElement.innerHTML = '';
+    }
+
+    GoogleBlockly.utils.dom.createSvgElement(
+      'rect',
+      {
+        fill: color.neutral_dark90,
+        x: 1,
+        y: 1,
+        width: FIELD_WIDTH,
+        height: FIELD_HEIGHT,
+        rx: 3,
+      },
+      this.backgroundElement
+    );
+
+    const graphNotes: TuneGraphEvent[] = generateGraphDataFromTune({
+      tuneEventValue: this.getValue(),
+      width: FIELD_WIDTH,
+      height: FIELD_HEIGHT,
+      numOctaves: 3,
+      startOctave: 4,
+      padding: 2,
+      noteHeightScale: 4,
+    });
+
+    graphNotes.forEach(graphNote => {
+      GoogleBlockly.utils.dom.createSvgElement(
+        'rect',
+        {
+          fill: '#68d1f7',
+          x: graphNote.x,
+          y: graphNote.y,
+          width: graphNote.width,
+          height: graphNote.height,
+          rx: 1,
+        },
+        this.backgroundElement
+      );
+    });
+
+    this.renderContent();
+  }
+
+  updateSize_() {
+    const width = FIELD_WIDTH + 2 * FIELD_PADDING;
+    const height = FIELD_HEIGHT + 2 * FIELD_PADDING;
+
+    this.borderRect_?.setAttribute('width', '' + width);
+    this.borderRect_?.setAttribute('height', '' + height);
+
+    this.size_.width = width;
+    this.size_.height = height;
+  }
+
+  protected showEditor_() {
+    super.showEditor_();
+
+    const editor = this.createDropdown();
+    DropDownDiv.getContentDiv().appendChild(editor);
+
+    const style = (this.sourceBlock_ as BlockSvg).style;
+    DropDownDiv.setColour(style.colourPrimary, style.colourTertiary);
+
+    DropDownDiv.showPositionedByField(this, this.disposeDropdown.bind(this));
+  }
+
+  private createDropdown(): HTMLDivElement {
+    this.newDiv = document.createElement('div');
+
+    this.renderContent();
+
+    this.newDiv.style.color = color.neutral_light;
+    this.newDiv.style.width = 'auto';
+    this.newDiv.style.backgroundColor = color.dark_black;
+    this.newDiv.style.padding = '5px';
+
+    return this.newDiv;
+  }
+
+  private renderContent(): void {
+    if (!this.newDiv) {
+      return;
+    }
+
+    ReactDOM.render(
+      React.createElement<TunePanelProps>(TunePanel, {
+        library: this.options.getLibrary(),
+        initValue: this.getValue(),
+        onChange: this.onValueChange,
+        ...this.options,
+      }),
+      this.newDiv
+    );
+  }
+
+  private disposeDropdown() {
+    this.newDiv = null;
+  }
+
+  private getTruncatedNotes(notes: number[]): string {
+    const allNotes = notes
+      .map(note => getNoteName(note))
+      .slice(0, MAX_DISPLAY_NOTES)
+      .join(', ');
+    return notes.length > MAX_DISPLAY_NOTES ? allNotes + '...' : allNotes;
```

Comment:
I don't think we should open this can of worms atm, but just noting that the hardwired ellipsis here may cause localization issues. It's not universally used in all languages, and in RTL languages this would end up showing up on the wrong side of the list.

I'm not sure the best approach. Break it out into a formatted template that can be rtl'ed? make it an actually localizable string? Use the unicode glyph? I dunno.

Here's a little info:
https://stackoverflow.com/questions/48718469/how-is-ellipsis-localized

Prob reasonable to defer, but wanted to note it.